.github/workflows/on_pull_request.yml

```diff
@@ -20,56 +18,14 @@ concurrency:
 
 jobs:
   onPullRequestJob:
+    env:
+      ANDROID_HOME: ${{ github.workspace }}/bin/androidSdk
     name: Verify code base when pull request is published/updated
-    runs-on: macos-latest
-
+    runs-on: ubuntu-latest
+    #    runs-on: ubuntu-20.04-16core # Larger github runner, with KVM acceleration
     steps:
-      - name: Run checkout github action
-        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # pin@v4
-        with:
-          lfs: 'true'
-          fetch-depth: 0
-
-      - name: Detect Arch
-        id: detect-arch
-        uses: ./config/actions/detect-arch
-
-      - name: Setup GitHub Runner workflow
-        uses: ./config/actions/setup-runner
-
-      - name: Get latest tag
-        id: latest-tag
-        uses: ./config/actions/get-latest-tag
-        with:
-          pattern: 'v*'
-
-      - name: Generate version code
-        id: version-code
-        uses: ./config/actions/generate-version-code
-
-      - name: Generate version name
-        id: version-name
-        uses: ./config/actions/version-name
-        with:
-          version-code: ${{ steps.version-code.outputs.version-code }}
-          version-name: ${{ steps.latest-tag.outputs.current-tag }}
-
-      - name: Run gradle instrumentation tests
-        uses: ./config/actions/gradle-connected-test
+      - name: Run pull request workflow
+        uses: govuk-one-login/mobile-android-pipelines@217b8fb9b8f5f1e763e8d56427d68180d422d62b
```

Comment:
I'm not a huge fan of this. Do we not think a better pattern might be to have each of the consumers create and manage their own workflow files, making use of the action files here?

Reply:
The aim really was to enforce our process consistently across our repositories; i.e. if we want to add a new step, we can add it once and it gets made available everywhere.

Reply:
as discussed, we can parameterise any differences between repos